hack/e2e/kops.sh

```diff
@@ -88,36 +93,63 @@ function kops_delete_cluster() {
   ${BIN} delete cluster --name "${CLUSTER_NAME}" --state "${KOPS_STATE_FILE}" --yes
 }
 
-# TODO switch this to python or work exclusively with yaml, all this
-# hacking with jq stinks!
+# TODO switch this to python, work exclusively with yaml, use kops toolbox
+# template/kops set?, all this hacking with jq stinks!
 function kops_patch_cluster_file() {
-  CLUSTER_FILE=${1}    # input must be json
+  CLUSTER_FILE=${1}    # input must be yaml
   KOPS_PATCH_FILE=${2} # input must be yaml
+  KIND=${3}            # must be either Cluster or InstanceGroup
+  ROLE=${4}            # must be either Master or Node
 
   loudecho "Patching cluster $CLUSTER_NAME with $KOPS_PATCH_FILE"
 
-  # Temporary intermediate files for patching
+  # Temporary intermediate files for patching, don't mutate CLUSTER_FILE until
```

Comment:
Does this create the cluster.yaml file before the test?  Is this something we can use `KOPS_FEATURE_FLAGS="SpecOverrideFlag"` for?  Also, are these changes in kops.sh necessary for this PR?

Reply:
cluster.yaml gets created by the dry run create somewhere else. if it already exists, the pathcing won't hapepn.

No these changes aren't necessary, I just wanted to piggyback the changes from fsx repo

Reply:
to be clear the pr desription is only relevant for first commit.

commit 2 and 3 are snuck in

Reply:
Ok, might want to break it up or update the PR description.